src/CommunityToolkit.Maui.Core/Essentials/FolderPicker/FolderPickerImplementation.windows.cs

```diff
@@ -0,0 +1,42 @@
+using System.Diagnostics;
+using CommunityToolkit.Maui.Core.Primitives;
+using Windows.Storage.Pickers;
+
+namespace CommunityToolkit.Maui.Storage;
+
+/// <inheritdoc />
+class FolderPickerImplementation : IFolderPicker
+{
+	/// <inheritdoc />
+	public async Task<Folder> PickAsync(string initialPath, CancellationToken cancellationToken)
+	{
+		cancellationToken.ThrowIfCancellationRequested();
+		var folderPicker = new Windows.Storage.Pickers.FolderPicker()
+		{
+			SuggestedStartLocation = PickerLocationId.DocumentsLibrary
+		};
+		WinRT.Interop.InitializeWithWindow.Initialize(folderPicker, Process.GetCurrentProcess().MainWindowHandle);
+		folderPicker.FileTypeFilter.Add("*");
+		var folderPickerOperation = folderPicker.PickSingleFolderAsync();
+
+		void CancelFolderPickerOperation()
+		{
+			folderPickerOperation.Cancel();
+		}
+
+		cancellationToken.Register(CancelFolderPickerOperation);
```

Comment:
We need to Dispose of the cancellationTokenRegistration, the return type of `Register` method
```suggestion
		var ctsRegistration = cancellationToken.Register(CancelFolderPickerOperation);
		//...
		ctsRegistration.Dispose();
```